src/job.rs

```diff
@@ -35,8 +35,13 @@ use crate::{config::Config, constants::Blocked};
 use crate::{constants::Chattype, contact::Contact};
 use crate::{scheduler::InterruptInfo, sql};
 
-// results in ~3 weeks for the last backoff timespan
-const JOB_RETRIES: u32 = 17;
+#[non_exhaustive]
+struct JobRetries;
+
+impl JobRetries {
+    pub const GENERAL: u32 = 17;
+    pub const SMTP_ERROR_TRANSIENT: u32 = 0;
```

Comment:
Why 0?

Reply:
Because it makes no sense to retry something that will fail anyways.

Reply:
But maybe it will not fail (transient means temporary). Didn't we want to put this to 3? (although I still think that 3 is quite low, as if we set this value to 3, then DC will retry once after 30s, once after 1min, once after 2min and then the message will directly be failed, just because the user had no internet for 2min after hitting the send button)

Reply:
But if the message fails because of no internet, the job doesn't get a transient error so it would try more then 3 times. 
We could match more exactly the transient dns errors and let them fail directly instead of letting all transient errors fail after 1 try. But still I think there's no sense in retrying dns errors.

Reply:
> But if the message fails because of no internet, the job doesn't get a transient error so it would try more then 3 times.

oops, you're right :)

> But still I think there's no sense in retrying dns errors.

If the server tells us that the error is transient, then maybe the server is actually right, e.g. because of a short server outage (above, with `maybe_transient`, we actually decided that one special "permanent" error is actually transient and should not be retried).

`self.tries >= JobRetries::SMTP_ERROR_TRANSIENT` is _always_ true currently (this is also why `#[allow(clippy::absurd_extreme_comparisons)]` was needed).

Reply:
> If the server tells us that the error is transient, then maybe the server is actually right, e.g. because of a short server outage 

Claro, but maybe the server is also wrong and a DNS error is something that is not getting fixed in a short time and block the user from sending any message to a group/contact.

> (above, with maybe_transient, we actually decided that one special "permanent" error is actually transient and should not be retried).

I think it's the other way around, we decided that one permanent error is actually transient and should get retried ;) Otherwise we wouldn't want to retry it.

Reply:
BTW, even the RFC for "Enhanced Mail System Status Codes" states for the `x.1.2` error following:
> The destination system specified in the address does not
> exist or is incapable of accepting mail.  For Internet mail
> names, this means the address portion to the right of the
> "@" is invalid for mail.  This codes is only useful for
> permanent failures.

"This codes is only useful for permanent failures". So I propose to change to code to just catch this error code and let this fail permanently.

Reply:
There are other types of transient errors, and even DNS failure may be temporary (SMTP server's local DNS server failed, dnsmasq was rebooting or something like this).